File: 461. Hamming Distance.java
// 461. Hamming Distance
// Time: O(1)
// Space: O(1)

// traverse bit and compare
public class Solution {
    public int hammingDistance(int x, int y) {
        int dis = 0;
        while (x != 0 || y != 0) {
            if ((x & 1) != (y & 1)) {
                dis += 1;
            }
            x >>>= 1;
            y >>>= 1;
        }
        return dis;
    }
}

File: 461. Hamming Distance.py
# 461. Hamming Distance
# Time: O(len(binary digits))
# Space: O(1)

# use XOR operation to get the diff digits and shift bits
class Solution:
    def hammingDistance(self, x: 'int', y: 'int') -> 'int':
        xor = x ^ y
        cnt = 0
        while xor != 0:
            if (xor & 1) == 1:
                cnt += 1
            xor >>= 1
        return cnt
